### PR TITLE
dbeaver/dbeaver#22103 replace autosave focus listener with dispose & tabbing support for datetime

### LINKED ISSUE #22103: tab away from datetime field doesn't work as expected
**Is your feature request related to a problem? Please describe.**
When I am editing a datetime field, and I tab away from it, I am taken first to the "value" tab button.  A second tab takes me to the value input field.  This is different than what happens in a string field, in which a tab acts as it does in excel and moves me to the next field.  If I continue tabbing, I end up with a null in the date field.  The only way to preserve the value that I have entered in the date field is to move my mouse and click on a different field.

**Describe the solution you'd like**
I believe that tabbing away from a date field should act the same as tabbing away from a text field

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/dbeaver#22103
Removes hacky way of autosaving for Linux, so it should be tested #10561
*Needs to be tested on MacOS and Linux*
Focus listener is unfortunately not viable for supporting traverse inside the multi control composites, so it was replaced with dispose listener.

Test that all values save properly after inline editing

### CODE REVIEW

**Comment by MashaKorax** [comment]
**Mac issue** : if the cell is in edit mode and open any application or editor over the app, it becomes full screen
**Common issue:** if the cell is in edit mode and go to the value panel, make changes there - nothing happens when you press the apply button, but only when the focus changes 

https://github.com/dbeaver/dbeaver/assets/84867187/ce556226-41e8-4d0c-a0ea-216afed2b311

**Comment by MashaKorax** [comment]
value panel is not refreshed
https://github.com/dbeaver/dbeaver/assets/84867187/aa0220f1-89d7-4140-8769-7e01e472177f

**Comment by MashaKorax** [comment]
This case is broken again https://github.com/dbeaver/dbeaver/issues/10561

**Comment by MashaKorax** [comment]
https://github.com/dbeaver/dbeaver/issues/10561 broken on linux

**Comment by MashaKorax** [comment]
verified

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/spreadsheet/Spreadsheet.java [hunk 1]
```diff
@@ -370,6 +374,9 @@ public void handleEvent(final Event event) {
                 // we don't want to mess current grid state
                 UIUtils.asyncExec(() -> presentation.navigateLink((GridCell) event.data, event.stateMask));
                 break;
+            case SWT.MouseExit:
+                System.out.println("Spreadsheet.handleEvent()");
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/spreadsheet/Spreadsheet.java]: Please remove

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/spreadsheet/Spreadsheet.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/spreadsheet/Spreadsheet.java]: fixed

**Pull request closed without merging**